### PR TITLE
Add scaling factors for ASTRA >= 1.9.9.dev

### PR DESCRIPTION
**Pull request by wjp**
This adapts the ODL astra_cuda backend scaling factors to the new adjoint scaling in astra-1.9.9.dev (current astra-toolbox master since a few days ago, astra-toolbox/astra-toolbox@54af7e8e22a3f1c9d90b13291b28d39778c05564), as announced in #1415 .

It is probably possible to clean things up further now, using a feature flag for example, but I'll leave that to you.

### CODE REVIEW

File: odl/tomo/backends/astra_cuda.py [hunk 1]
```diff
@@ -374,7 +375,10 @@ def astra_cuda_bp_scaling_factor(proj_space, reco_space, geometry):
     scaling_factor /= (reco_space.weighting.const /
                        reco_space.cell_volume)
 
-    if parse_version(ASTRA_VERSION) < parse_version('1.8rc1'):
+    if parse_version(ASTRA_VERSION) >= parse_version('1.9.9.dev'):
+        scaling_factor /= float(reco_space.cell_volume)
+        scaling_factor *= float(geometry.det_partition.cell_volume)
+    elif parse_version(ASTRA_VERSION) < parse_version('1.8rc1'):
```
Comment (adler-j) [inline odl/tomo/backends/astra_cuda.py]: Prefer consistent directions (e.g. `>` everywhere or `<` everywhere), it's a bit hard to follow the if statements atm.

Comment (wjp) [reply odl/tomo/backends/astra_cuda.py]: Fair enough. I've added a separate commit to make things chronological.

I didn't do this before because the version currently in master is non-chronological, which made for a confusing diff. This separate commit to clean this up is hopefully the best of both worlds.

File: odl/tomo/backends/astra_cuda.py [hunk 2]
```diff
@@ -463,10 +464,10 @@ def astra_cuda_bp_scaling_factor(proj_space, reco_space, geometry):
             # missing factor src_radius ** 2 in the ASTRA BP with
             # density weighting.
             det_px_area = geometry.det_partition.cell_volume
-            scaling_factor *= (src_radius ** 2 * det_px_area ** 2 /
-                               reco_space.cell_volume ** 2)
-
-        # TODO: add case with new ASTRA release
+            scaling_factor *= (src_radius ** 2 * det_px_area ** 2)
+    else:
```
Comment (adler-j) [inline odl/tomo/backends/astra_cuda.py]: Scary catch-all else statement. Is this supposed to catch all above something? In that case be explicit about that and lets have a else with an error message.

Comment (adler-j) [reply odl/tomo/backends/astra_cuda.py]: Could we just write that explicitly instead? Just easier to read (and less likely to be broken by some weird mistake later on). After that we merge.

Comment (wjp) [reply odl/tomo/backends/astra_cuda.py]: Sure. I've added explicit comments about the version ranges to all four cases for consistency.

**Comment by adler-j** [comment]
Ready to merge?

File: odl/tomo/backends/astra_cuda.py (hunk 2 above)
Comment (kohr-h) [reply odl/tomo/backends/astra_cuda.py]: IMO the catch-all `else:` doesn't really do any harm. If it were replaced with an `elif ... >1.9.9dev` and we somehow screwed up such that none of the cases applied, the function would still return a (most likely wrong) value.

The very cautious version would be the `elif ... >1.9.9dev` case, plus an `else:` case that raises a `RuntimeError`. But maybe normal reasoning is good enough here.

**Comment by wjp** [comment]
Since it is also related to the 1.9.9.dev upgrade, I've also added the fix for the issue identified in #1529 directly to this PR. This should now cover the full `consistent_scaling` branch merge on the ASTRA side.
